lib/scraper.py

```diff
@@ -1,5 +1,40 @@
-from utils import Stock
+from typing import List
+import requests
+import re
+
+from bs4 import BeautifulSoup
+from lib.utils.classes import ChronologicalFloat, Stock
+from lib.utils.args import debug
+from bs4 import BeautifulSoup, Tag
 
 # Given a ticker, scrapes the web for information and returns a Stock object
-def scrape(ticker: str) -> Stock:
-    raise Exception("scraper.scrape: Not implemented")
+# Ticker: pd.DataFrame: Name, Ticker, URL
+def scrape(session: requests.Session, ticker: str) -> Stock:
+    debug("Scraping %s" % ticker)
+    url = "https://www.screener.in/company/%s/consolidated/" % ticker
+    page = session.get(url)
+    soup = BeautifulSoup(page.text, 'html.parser')
+    name = soup.find('h1').text
+    [net_profit] = parse_table(soup.select_one('#profit-loss .data-table'), ['net profit'])
+
+    stock = Stock(ticker, name, url, net_profit)
+    return stock
```

Comment:
Instead of having individual metrics as args, we can declare a dataframe and update that.

Reply:
Yeah that seems like the way to go